tests/unit/test_toolchain.cpp

```diff
@@ -34,24 +34,32 @@
 
 void test_packed1()
 {
-    typedef OT_TOOL_PACKED_BEGIN struct
+    OT_TOOL_PACKED_BEGIN
+    struct packed_t
     {
         uint8_t  mByte;
         uint32_t mWord;
         uint16_t mShort;
-    } OT_TOOL_PACKED_END packed_t;
+    } OT_TOOL_PACKED_END;
 
+#if _WIN32
+    static_assert(sizeof(packed_t) == 7, "Toolchain::OT_TOOL_PACKED failed 1");
+#endif
     VerifyOrQuit(sizeof(packed_t) == 7, "Toolchain::OT_TOOL_PACKED failed 1\n");
 }
 
 void test_packed2()
 {
-    typedef OT_TOOL_PACKED_BEGIN struct
+    OT_TOOL_PACKED_BEGIN
```

Comment:
There is, but it is in direct contradiction to one of the other platforms (I don't remember which). This was the only truly generic way to get it to compile.